src/lib_io.c

```diff
@@ -164,8 +164,14 @@ static void io_file_readall(lua_State *L, FILE *fp)
   MSize m, n;
   for (m = LUAL_BUFFERSIZE, n = 0; ; m += m) {
     char *buf = lj_buf_tmp(L, m);
-    n += (MSize)fread(buf+n, 1, m-n, fp);
-    if (n != m) {
+    for (;;) {
+      n += (MSize)fread(buf+n, 1, m-n, fp);
+      if (n == m) break;
+      else if (ferror(fp) != 0) {
+        if (errno != EINTR) return;
```

Comment:
I wonder if we should use `break` instead of `return` here.

Reply:
Hmm, `break` is not right either. Better return the partially read data here before returning?

Reply:
Do we need to? It seems to me like the only code path we are following after this is [luaL_fileresult](https://github.com/openresty/luajit2/blob/master/src/lib_aux.c#L31) which returns `nil, errno, string` to the Lua-land? We also avoid repeating the GC step).

Reply:
On second thought, maybe we do need the GC step anyway for the buffer allocated above. I originally wasn't sure it was needed, but I do think so now.